src/infrastructure/controller/AccountController.ts

```diff
@@ -0,0 +1,166 @@
+import { Request, Response } from 'express'
+import * as AccountService from '@service/AccountService'
+import ResourceNotFoundError from '@error/ResourceNotFoundError'
+import InvalidAccountOperationError from '@error/InvalidAccountOperationError'
+import AccountType from '@model/account/AccountType'
+import { port } from '@entrypoint'
+import InvalidAccountFilterError from '@error/InvalidAccountFilterError'
+
+const extractUUIDs = (req: Request) => [req.params.userUUID, req.params.accUUID]
+
+export const openAccount = async (req: Request, res: Response) => {
+  const userUUID = req.params.userUUID
+  const accountType = req.body.accountType
+  try {
+    const accUUID = await AccountService.openAccount(userUUID, accountType)
+    const resourcePath = `http://${req.hostname}:${port}/user/${userUUID}/account/${accUUID}`
+    res.status(201).location(resourcePath).send()
```

Comment:
In spring hateoas module, there are some cool utilites to automatically generate urls to resources, based on references to controller methods. Is there something similar in express, or does it always have to be done like I did here?